packages/flutter/lib/src/material/ink_sparkle.dart

```diff
@@ -0,0 +1,3667 @@
+          weight: 60,
+        ),
+      ],
+    ).animate(_animationController);
+
+    _sparkleAlpha = TweenSequence<double>(
+      <TweenSequenceItem<double>>[
+        TweenSequenceItem<double>(
+          tween: Tween<double>(begin: 0.0, end: 1.0),
+          weight: 13,
+        ),
+        TweenSequenceItem<double>(
+          tween: ConstantTween<double>(1.0),
+          weight: 27,
+        ),
+        TweenSequenceItem<double>(
+          tween: Tween<double>(begin: 1.0, end: 0.0),
+          weight: 50,
+        ),
+      ],
+    ).animate(_animationController);
+
+    // Creates an element of randomness so that ink eminating from the same
+    // pixel have slightly different rings and sparkles.
+    _turbulenceSeed = turbulenceSeed ?? math.Random().nextDouble() * 1000.0;
+  }
+
+  static const Duration _animationDuration = Duration(milliseconds: 617);
+  static const double _targetRadiusMultiplier = 2.3;
+  static const double _rotateRight = math.pi * 0.0078125;
+  static const double _rotateLeft = -_rotateRight;
+  static const double _noiseDensity = 2.1;
+
+  late AnimationController _animationController;
+
+  // The Android 12 version has these values calculated in the GLSL. They are
+  // constant for every pixel in the animation, so the Flutter implementation
+  // computes these animation values in software in order to simplify the shader
+  // implementation and provide better performance on most devices.
+  late Animation<Vector2> _center;
+  late Animation<double> _radiusScale;
+  late Animation<double> _alpha;
+  late Animation<double> _sparkleAlpha;
+
+  late double _turbulenceSeed;
+
+  final Color _color;
+  final Offset _position;
+  final BorderRadius _borderRadius;
+  final ShapeBorder? _customBorder;
+  final double _targetRadius;
+  final RectCallback? _clipCallback;
+  final TextDirection _textDirection;
+
+  /// Used to specify this type of ink splash for an [InkWell], [InkResponse],
+  /// material [Theme], or [ButtonStyle].
+  ///
+  /// Since no [turbulenceSeed] is passed, the effect will be random for
+  /// subsequent presses in the same position.
+  static const InteractiveInkFeatureFactory splashFactory = _InkSparkleFactory();
+
+  /// Used to specify this type of ink splash for an [InkWell], [InkResponse],
+  /// material [Theme], or [ButtonStyle].
+  ///
+  /// Since a [turbulenceSeed] is passed, the effect will not be random for
+  /// subsequent presses in the same position. This can be used for testing.
+  static const InteractiveInkFeatureFactory constantTurbulenceSeedSplashFactory = _InkSparkleFactory.constantTurbulenceSeed();
+
+  @override
+  void dispose() {
+    _animationController.stop();
+    _animationController.dispose();
+    super.dispose();
+  }
+
+  @override
+  void paintFeature(Canvas canvas, Matrix4 transform) {
+    // InkSparkle can only paint if its shader has been compiled.
+    if (_InkSparkleFactory._shaderManager == null) {
+      debugPrint(
```

Comment:
Typically, `debugPrint` should inly be called within an assert or guarded with `kDebugMode` / `kProfileMode`. Should we do this here as well? There's probably no point in printing this in release builds...

Reply:
changed to assert